backend/src/main/java/org/eclipse/tractusx/puris/backend/production/controller/ProductionController.java

```diff
@@ -0,0 +1,275 @@
+            @ApiResponse(responseCode = "404", description = "Planned Production does not exist."),
+            @ApiResponse(responseCode = "500", description = "Internal Server Error.")
+    })
+    @ResponseStatus(HttpStatus.OK)
+    public ProductionDto updateProduction(@RequestBody ProductionDto dto) {
+        OwnProduction updatedProduction = productionService.update(convertToEntity(dto));
+        if (updatedProduction == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Production does not exist.");
+        }
+        return convertToDto(updatedProduction);
+    }
+
+    @DeleteMapping("{id}")
+    @Operation(summary = "Deletes a planned production by its UUID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Planned Productions was deleted."),
+            @ApiResponse(responseCode = "400", description = "Malformed or invalid request body."),
+            @ApiResponse(responseCode = "404", description = "Planned Production does not exist."),
+            @ApiResponse(responseCode = "500", description = "Internal Server Error.")
+    })
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteProduction(@PathVariable UUID id) {
+        OwnProduction production = productionService.findById(id);
+        if (production == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Production does not exist.");
+        }
+        productionService.delete(id);
+    }
+
+    @GetMapping("partner")
+    @ResponseBody
+    @Operation(summary = "Get all productions of partners for a material", description = "Get all productions of partners for a material number. Optionally the partners can be filtered by their bpnl and the production site can be filtered by its bpns.")
+    public List<ProductionDto> getAllProductionsForPartner(String materialNumber, Optional<String> bpnl,
+            Optional<String> site) {
+        if (bpnl.isEmpty()) {
+            if (site.isEmpty()) {
+                return partnerProductionService.findAll().stream()
+                        .filter(prod -> prod.getMaterial().getOwnMaterialNumber().equals(materialNumber))
+                        .map(this::convertToDto)
+                        .collect(Collectors.toList());
+            }
+            return partnerProductionService.findAll().stream()
+                    .filter(prod -> prod.getMaterial().getOwnMaterialNumber().equals(materialNumber) && prod.getProductionSiteBpns().equals(site.get()))
+                    .map(this::convertToDto).collect(Collectors.toList());
+        }
+
+        Partner partner = partnerService.findByBpnl(bpnl.get());
+        if (partner == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Partner does not exist.");
+        }
+        return partnerProductionService.findAllByPartnerId(partner.getUuid()).stream()
```

Comment:
I guess you just went the same way as we did previously. If more convenient, feel free to create a suiting method in productionService